src/client/lib/_mapbox/map-factory.js

```diff
@@ -0,0 +1,49 @@
+import BaseLayerControl from './baselayer-control'
+import mapboxgl from './_mapbox'
+import { MAP_CENTER, MAP_ZOOM, MAP_BASELAYERS, MAP_BASELAYER_DEFAULT } from './map-config'
+
+import baselayerControlComponent from './baselayer-control-component';
+
+export default function(container) {
+  const mapLayers = []
+
+  const map = new mapboxgl.Map({
+    container,
+    center: MAP_CENTER,
+    zoom: MAP_ZOOM,
+    style: MAP_BASELAYER_DEFAULT.style
+  })
+
+  const _addLayer = map.addLayer.bind(map)
+  map.addLayer = (layer, before) => {
+    mapLayers.push(layer)
+    _addLayer(layer, before)
+  }
+
+  const _removeLayer = map.removeLayer.bind(map)
+  map.removeLayer = (id) => {
+    mapLayers = mapLayers.filter((layer) => layer.id !== id)
+    _removeLayer(layer, before)
+  }
```

Comment:
you’re redeclaring these functions so that they know about the layers of the map right? Maybe  collecting these functions on top with a prefix explicitly stating their source will add clarity. Add a comment on why you’re remapping them